### PR TITLE
Added bindings for some vital types

### PR DESCRIPTION
**Pull request by massir-rpi**
Bindings added for landmark_map, bounding_box<int>, match_set

### CODE REVIEW

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by tao558** [comment]
@as6520 What are your thoughts on merging this into dev/finish_vital_type_bindings? We'd have all of the vital type bindings in one place.

File: python/kwiver/vital/types/covariance_class.h [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+: public PyCovarianceBase
+{
+
+  kwiver::vital::covariance_<2, double> covar;
+
+  public:
+
+    PyCovariance2d() {covar = kwiver::vital::covariance_<2, double>();};
+    PyCovariance2d(double mat)
+                  {covar = kwiver::vital::covariance_<2, double>(mat);};
```
Comment (linus-sherrill) [inline python/kwiver/vital/types/covariance_class.h:68]: This file does not abide by the style manual with regard to indenting. Also private members usually follow the public members, even though this is not in the style manual.

File: python/kwiver/vital/types/covariance_class.cxx [hunk 2]
```diff
@@ -286,4 +120,4 @@ covar_from_matrix(int N, char c_type, py::object init)
     retVal = std::shared_ptr<PyCovarianceBase>(new PyCovariance3f(mat));
   }
   return retVal;
-}
+}
```
Comment (linus-sherrill) [inline python/kwiver/vital/types/covariance_class.cxx:123]: Files should end with a newline. This also applies to other files.

File: python/kwiver/vital/types/landmark.h [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    virtual void set_color(kwiver::vital::rgb_color) { };
+
+    virtual unsigned get_observations() { return 0;};
+    virtual void set_observations(unsigned) { };
+};
+
+class PyLandmarkd
+: public PyLandmarkBase
+{
+
```
Comment (linus-sherrill) [inline python/kwiver/vital/types/landmark.h:74]: This file does not abide by the style manual with regard to indenting. Also private members usually follow the public members, even though this is not in the style manual.

**Comment by as6520** [comment]
@tao558 I agree that the type bindings should be moved dev/finish_vital_type_bindings. Based on the offline discussion, we can move the commits to correct branch and then land it with all the other types bindings with the requested changes

**Comment by tao558** [comment]
> @tao558 I agree that the type bindings should be moved dev/finish_vital_type_bindings. Based on the offline discussion, we can move the commits to correct branch and then land it with all the other types bindings with the requested changes

@as6520 just to be clear, you're saying merge into that branch now and deal with any required changes when we merge that branch? I'd agree.

@linus-sherrill are you OK with this?

**Comment by as6520** [comment]
> @as6520 just to be clear, you're saying merge into that branch now and deal with any required changes when we merge that branch?
Based on my understanding, #942 was branched from dev/bind-algorithms. So merging it as is might merge those changes to dev/finish_vital_type_bindings. Since there is only 1 commit with all the types we can either cherry pick the commit or rebase onto the dev/finish_vital_type_bindings class. Then the changes that @linus-sherrill had requested would be addressed in the dev/finish_vital_type_bindings and this PR would be closed.

**Comment by tao558** [comment]
> > @as6520 just to be clear, you're saying merge into that branch now and deal with any required changes when we merge that branch?
> 
> Based on my understanding, #942 was branched from dev/bind-algorithms. So merging it as is might merge those changes to dev/finish_vital_type_bindings. Since there is only 1 commit with all the types we can either cherry pick the commit or rebase onto the dev/finish_vital_type_bindings class. Then the changes that @linus-sherrill had requested would be addressed in the dev/finish_vital_type_bindings and this PR would be closed.

I think cherry-picking would make sense here

**Pull request closed without merging**